processing/src/main/java/org/apache/druid/java/util/metrics/CgroupMemoryMonitor.java

```diff
@@ -58,11 +76,28 @@ public CgroupMemoryMonitor()
 
   @Override
   public boolean doMonitor(ServiceEmitter emitter)
+  {
+    if (isRunningOnCgroupsV2) {
+      return cgroupV2MemoryMonitor.doMonitor(emitter);
+    } else {
+      return doMonitorInternal(emitter, cgroupDiscoverer, dimensions, MEMORY_USAGE_FILE, MEMORY_LIMIT_FILE, this);
+    }
+  }
+
+  public static boolean doMonitorInternal(
```

Comment:
short javadoc might be helpful not that this is a static utility method used by v1 and v2 monitor. I don't fully understand the "internal" part of the name either